### PR TITLE
Have a real fixed HostID on Linux

### PR DESCRIPTION
**Pull request by pierresouchay**
On Linux, most golang programs do not run as root (or at least, they should not),
by default, the kernels uses strict permissions, so most userland programs cannot
read `/sys/class/dmi/id/product_uuid`. However, programs such as Consul are relying
on it to get fixed IDs, instead they have a different ID on each boot.

We propose to use `/etc/machine-id` as fallback https://www.freedesktop.org/software/systemd/man/machine-id.html

In order to fix this, this patch does the following:
 - if `/sys/class/dmi/id/product_uuid` can be read, use it for HostID
 - else if `/etc/machine-id` exists and has 32 chars, use it and add '-' to have the same format as product_uuid
 - finally, if notthing works, use the `kernel.random.boot_id`

This will greatly increase the number of programs having correct behaviour when
those rely on having a fixed HostID.

This will fix the following issues:
 - https://github.com/shirou/gopsutil/issues/350
 - https://github.com/hashicorp/consul/issues/4741

### CODE REVIEW

File: host/host_linux.go [hunk 1]
```diff
@@ -73,14 +73,24 @@ func InfoWithContext(ctx context.Context) (*InfoStat, error) {
 	}
 
 	sysProductUUID := common.HostSys("class/dmi/id/product_uuid")
+	machineID := common.HostEtc("machine-id")
 	switch {
+	// When not running as root, lib cannot read the value
```
Comment (Lomanic) [inline host/host_linux.go]: This is not the only way gopsutil could not be able to read this value, the file can be missing for example (this is the case on arm on Debian 9). Remove this comment or completely rephrase it to explain what this `switch case fallthrough` construction does.

**Comment by pierresouchay** [comment]
@lomanic yes, but this case is already properly handled with error management below. 

What I mean is, thus case (file not being readable is not the exception: some weird kernel config or unusual arch), but the rule (I mean, only programs running as root can use it, which in 2018 should be the exception, not the rule)

**Comment by pierresouchay** [comment]
@Lomanic Thank you for the review

**Comment by shirou** [comment]
Wow, Great! Your are thinking backward compatibility so much!

**Comment by pierresouchay** [comment]
@shirou Thank you! Do you think you might release a new version, so I can fix the bug in Consul by updating this library? Kind Regards

**Comment by Lomanic** [comment]
Releases are once per month, following [calver](http://calver.org/) like Ubuntu, see [Tag semantics](https://github.com/shirou/gopsutil#tag-semantics)

**Comment by linydquantil** [comment]
+1
